### PR TITLE
Automatically warn a librarianette if a content warning is missing

### LINKED ISSUE #192: Auto populate content warnings
If a tag is marked as needing a content warning, automatically populate the Comments section with the CW

### PR DESCRIPTION
**Pull request by braxtonhall**
resolves #192

when #207 is merged this PR will make more sense to review, as that branch's changes have already been merged into here

Kapture 2023-01-05 at 19 14 57

### CODE REVIEW

File: src/ts/content/extensions/util/contentWarningCheck.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	tagsTextArea: HTMLTextAreaElement,
+	commentsTextArea: HTMLTextAreaElement
+) => {
+	const saveButtonListener = () => handleSave(tagsTextArea, commentsTextArea);
+	const showContentWarningCheck = () => onSave(saveButtonListener);
+	const hideContentWarningCheck = () => offSave(saveButtonListener);
+	return {showContentWarningCheck, hideContentWarningCheck};
+};
+
+export {appendContentWarningChecker};
```
Comment (adirar111) [inline src/ts/content/extensions/util/contentWarningCheck.ts:92]: *chef's kiss*

File: src/ts/content/extensions/util/contentWarningCheck.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+				],
+				colour: UIColour.PURPLE,
+			})
+		);
+	} else {
+		return true;
+	}
+};
+const getWarningRequiredTags = (commentsTextArea: HTMLTextAreaElement, tags: string[], trees: TagTrees): string[] => {
+	if (contentWarningIsPresent(commentsTextArea)) {
```
Comment (adirar111) [inline src/ts/content/extensions/util/contentWarningCheck.ts:57]: does this mean that if there's one `cw:` in the comments box, we don't show the warning?

Comment (adirar111) [reply src/ts/content/extensions/util/contentWarningCheck.ts:57]: this is reasonable, i'm just wondering

**Review by adirar111 — APPROVED** [review]
so sickkkkkkkkkkk

File: src/ts/content/extensions/util/contentWarningCheck.ts (hunk 2 above)
Comment (braxtonhall) [reply src/ts/content/extensions/util/contentWarningCheck.ts:57]: Yeah exactly. There just needs to be some content warning

**Pull request merged**